### DISCUSSION TITLE
Add a new leayer easily

### DISCUSSION THREAD

**Discussion by LeonardVertighel** [kimai/kimai#5140] [Questions & Answers]
Hi,
    we use Kimai and wanted to know if there is an easy way, such as using a plugin, to add a layer to the structure or do we need to implement it.
For example we would like a structure like this:
`Projects -> Activities -> New layer`
or
`Projects -> New layer -> Activities`
and the new layer needs some properties, e.g. name, description, code, ...

With the Custom Fields plugin you can add some fields, but I have not found anything to add a new level.

Is there any way other than implementing everything?

**Comment by kevinpapst** [comment]
No, there is nothing like that available. What do you have in mind?

**Comment by LeonardVertighel** [comment]
Sometimes we need to better detail the activities within a project. 

For example within a project we have 5 activities and 3 are directly related to each other and the other 2 are related to each other. 
We would need to identify the 2 groups of activities and assign them common characteristics.

A first simple implementation could be to use tags also for activities and not only for timesheets. It does not seem to me that there is an option to put tags in activities.

**Reply by kevinpapst** [reply]
The "custom fields" plugin has a feature called "auto tagging".
With that you can assign tags to activities. 
Whenever that activity is used in a timesheet, the activities tags are forced onto the timesheet. The user cannot remove them.
The same can be done with tags on project and customer level.

**Comment by LeonardVertighel** [comment]
Is it possible to display a field added with the “Custom field” plugin to the weekly hours page?

**Reply by kevinpapst** [reply]
No. This view cannot be extended.

**Reply by LeonardVertighel** [reply]
Hi,
        in reviewing the specifications with colleagues, it has become apparent that the link is not as previously indicated with the activities, but only with the project: the new level will have the same value as the activities.
So we would need to add a piece of information to the kimai structure. It is an element at the same level as the activities, related to a project. And it should also be possible to set it on the “Weekly hours” page with the dropdown in the same way as the activities. 
Logically, this is a non-mandatory element and can be included in the timesheet when it is created.
We have seen that you provide the ability to develop plugins for others. Would it be possible to take advantage of this possibility?
Is it possible to develop such a plugin?
If so, would it be necessary to send the specifications, perhaps more detailed, to email support@kimai.org, as mentioned on the page https://www.kimai.org/store/custom-plugins.html to get a price estimate?

**Reply by LeonardVertighel** [reply]
I will add some screenshots to clarify the idea
![NewLayer_3](https://github.com/user-attachments/assets/d01b1523-71e2-41d0-83fa-ef4cf5d5dc85)
![NewLayer_2](https://github.com/user-attachments/assets/32915d2b-c54a-4d17-a9d1-655c6c5b751f)
![NewLayer_1](https://github.com/user-attachments/assets/4453e79e-5478-45bb-96d8-40102adf07a2)

**Comment by LeonardVertighel** [comment]
I have sent an email to support@kimai.org with directions on how to develop the plugin, I look forward to your feedback.

Thank you
